util/src/main/java/org/killbill/billing/util/nodes/DefaultKillbillNodesService.java

```diff
@@ -96,6 +121,21 @@ public void start() {
     public void stop() {
         logger.info("Deleting nodeInfo for {}", CreatorName.get());
         nodeInfoDao.delete(CreatorName.get());
+        if (isStopped) {
+            return;
+        }
+        try {
+            nodeInfoExecutor.shutdown();
+            final boolean success = nodeInfoExecutor.awaitTermination(TERMINATION_TIMEOUT_SEC, TimeUnit.SECONDS);
+            if (!success) {
+                logger.warn("NodeInfoExecutor failed to complete termination within " + TERMINATION_TIMEOUT_SEC + "sec");
```

Comment:
You might want to try to `shutdownNow` here.

Reply:
We could, but again, this not not what we do in other part of the codes, e.g [here](https://github.com/killbill/killbill/blob/master/tenant/src/main/java/org/killbill/billing/tenant/api/TenantCacheInvalidation.java#L113). Also, if trying to shutdown cleanly fails (times-out), then is it important to try to force a shutdown of the pool knowing that shutdown sequence will continue -- and mostly timeout as a result and be aborted?